File: src/main/java/de/sean/splugin/spigot/events/LeaveEvent.java
package de.sean.splugin.spigot.events;

import de.sean.splugin.SPlugin;
import de.sean.splugin.discord.DiscordUtil;
import de.sean.splugin.util.SMessages;
import de.sean.splugin.util.SUtil;
import org.bukkit.ChatColor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class LeaveEvent implements Listener {
    @EventHandler
    public void PlayerLeave(PlayerQuitEvent event) {
        /* Format leave message */
        event.setQuitMessage(ChatColor.RED + SMessages.getRandomMessage("messages.leave").replace("[player]", event.getPlayer().getName()));
        
        /* AFK */
        SUtil.removeActivityForPlayer(event.getPlayer().getUniqueId());
        SUtil.removePlayerAFK(event.getPlayer().getUniqueId());

        /* Discord */
        final DiscordUtil discord = SPlugin.discord;
        if (discord.leaveMessage) {
            discord.sendMessage(SMessages.getRandomMessage("messages.leave").replace("[player]", event.getPlayer().getName()));
        }
    }
}

File: src/main/kotlin/de/sean/splugin/bukkit/events/LeaveEvent.kt
package de.sean.splugin.bukkit.events

import de.sean.splugin.SPlugin
import de.sean.splugin.util.SMessages.getRandomMessage
import de.sean.splugin.util.SUtil
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class LeaveEvent : Listener {
    @EventHandler
    fun playerLeave(event: PlayerQuitEvent) {
        /* Format leave message */
        event.quitMessage = getRandomMessage("messages.leave").replace("[player]", event.player.name)

        /* AFK */
        SUtil.playerLastActivity.remove(event.player.uniqueId)
        SUtil.afkPlayers.remove(event.player.uniqueId)

        /* Discord */
        val discord = SPlugin.discord
        if (discord.leaveMessage) {
            discord.sendMessage(getRandomMessage("messages.leave").replace("[player]", event.player.name))
        }
    }
}
